addons/purchase/models/purchase.py

```diff
@@ -613,6 +626,83 @@ def retrieve_dashboard(self):
 
         return result
 
+    def _send_reminder_mail(self):
+        template = self.env.ref('purchase.email_template_edi_purchase_reminder', raise_if_not_found=False)
+        if template:
+            mail_id = template.with_context(is_reminder=True).send_mail(self.id, raise_exception=False, notif_layout="mail.mail_notification_paynow")
+
+    @api.model
+    def send_reminder_mail(self):
+        if not self.user_has_groups('purchase.group_send_reminder') and not self.receipt_reminder_email:
+            return
+
+        for order in self._get_orders_to_remind():
+            date = order.date_planned or order.expected_date
+            if date and (date - relativedelta(days=order.reminder_date_before_receipt)).date() == datetime.today().date():
+                order._send_reminder_mail()
+
+    @api.model
+    def _get_orders_to_remind(self):
+        """When auto sending a reminder mail, only send for unconfirmed purchase
+        order and not all products are service."""
+        return self.search([
+            ('state', 'in', ['purchase', 'done']),
+            ('mail_reminder_confirmed', '=', False)
+        ]).filtered(lambda p: p.mapped('order_line.product_id.product_tmpl_id.type') != ['service'])
+
+    def get_confirm_url(self, confirm_type=None):
```

Comment:
this doesn't need to be public right ?

Reply:
I tried, but it seems the method used in template have to be public.

Reply:
ok, wasn't aware